### ISSUE TITLE
AMD CI tracking issue

### ISSUE THREAD

**Issue by ivarflakstad** [huggingface/transformers#36019]
Tracking issue for efforts related to CI running on AMD hardware.

Intended to record overall progress.

### Status

CI on AMD is running.
There are several different categories of failed tests that need to be addressed.

It's worth noting that many of these failed tests are not due to bugs, but rather that the test case is opinionated. For example a test can fail because the expected logits do not match the result, but different hardware can produce different logits and still be correct.
This is known to transformers and techniques to address this are already in place.
